apm-agent-plugins/apm-kafka-plugin/apm-kafka-spring-plugin/src/test/java/co/elastic/apm/agent/kafka/KafkaSpringIT.java

```diff
@@ -0,0 +1,227 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static co.elastic.apm.agent.testutils.assertions.Assertions.assertThat;
+
+
+public class KafkaSpringIT extends AbstractInstrumentationTest {
+
+    private static final String SINGLE_TOPIC = "single-topic";
+    private static final String BATCH_TOPIC = "batch-topic";
+
+    private static KafkaContainer kafka;
+
+    private static AnnotationConfigApplicationContext springContext;
+    private static KafkaTemplate<Integer, String> kafkaTemplate;
+
+    @BeforeClass
+    @SuppressWarnings("unchecked")
+    public static void setup() throws ClassNotFoundException {
+        // confluent versions 7.1.0 correspond Kafka versions 3.1.0 -
+        // https://docs.confluent.io/current/installation/versions-interoperability.html#cp-and-apache-ak-compatibility
+        kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka").withTag("7.1.0"));
+        kafka.withCreateContainerCmdModifier(TestContainersUtils.withMemoryLimit(4096));
+        kafka.start();
+
+        springContext = new AnnotationConfigApplicationContext(SpringConfig.class);
+        kafkaTemplate = (KafkaTemplate<Integer, String>) springContext.getBean("kafkaTemplate");
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        springContext.close();
+        kafka.stop();
+    }
+
+    @Test
+    public void testBatchReceive() {
+        Transaction transaction1 = startTestRootTransaction("Send 1");
+        kafkaTemplate.send(BATCH_TOPIC, "data-1");
+        transaction1.deactivate().end();
+
+        Transaction transaction2 = startTestRootTransaction("Send 2");
+        kafkaTemplate.send(BATCH_TOPIC, "data-2");
+        transaction2.deactivate().end();
+
+        reporter.awaitTransactionCount(3, 10000);
+        reporter.awaitSpanCount(3); //two kafka exit spans + the batchListener span
+
+        List<Span> senderSpans = reporter.getSpans().stream()
+            .filter(AbstractSpan::isExit)
+            .collect(Collectors.toList());
+        assertThat(senderSpans).hasSize(2);
+
+        Transaction batchReceive = reporter.getTransactions().stream()
+            .filter(transaction -> transaction.getNameAsString().equals("Spring Kafka Message Batch Processing"))
+            .findFirst().orElse(null);
+
+        assertThat(batchReceive)
+            .isNotNull()
+            .hasSpanLinkCount(2)
+            .hasSpanLink(senderSpans.get(0))
+            .hasSpanLink(senderSpans.get(1));
+
+        Span listenerSpan = reporter.getSpanByName("batchListener");
+        assertThat(listenerSpan)
+            .isNotNull()
+            .hasParent(batchReceive);
+    }
+
+    // This test has nothing to do with the SpringKafkaBatchListenerInstrumentation,
+    // it just verifies that non-batch listeners are caught by the standard kafka instrumentation
+    @Test
+    public void testSingleReceive() {
```

Comment:
[very minor] maybe rename the test to something like `testNonBatchSingleReceive` so it's a bit more explicit that it does not relate to spring batch instrumentation without reading the comment above.